rbx_binary/src/deserializer/error.rs

```diff
@@ -79,4 +79,11 @@ pub(crate) enum InnerError {
         expected_type_id: u8,
         actual_type_id: u8,
     },
+
+    #[error("Failed to deserialize {class_name}.{prop_name} because {source}")]
+    PropertyError {
```

Comment:
I'm inclined to agree that `PropertyError` is a bad name. Maybe something like `DataTypeError` instead? That would let us perhaps fold stuff like CFrame's `BadRotationId` error variant into it in the future.

Reply:
Renamed to `BadPropertyValue`, and we can replace usage of `BadRotationId` with it  (and possibly `InvalidPropData`)?